src/ScheduleFunctions.cpp

```diff
@@ -600,21 +603,31 @@ Stmt build_extern_produce(const map<string, Function> &env, Function f, const Ta
 
             mark_buffer = Block::make(mark_buffer, mark_shape);
 
-            if (!is_no_op(annotate)) {
-                annotate = Block::make(annotate, mark_buffer);
+            if (!is_no_op(pre_call)) {
+                pre_call = Block::make(pre_call, mark_buffer);
             } else {
-                annotate = mark_buffer;
+                pre_call = mark_buffer;
             }
         }
         for (const auto &buffer : buffers_contents_to_annotate) {
             // Return type is really 'void', but no way to represent that in our IR.
             // Precedent (from halide_print, etc) is to use Int(32) and ignore the result.
             Stmt mark_contents = Evaluate::make(
                 Call::make(Int(32), "halide_msan_annotate_buffer_is_initialized", {buffer}, Call::Extern));
-            if (!is_no_op(annotate)) {
-                annotate = Block::make(annotate, mark_contents);
+            if (!is_no_op(pre_call)) {
+                pre_call = Block::make(pre_call, mark_contents);
             } else {
-                annotate = mark_contents;
+                pre_call = mark_contents;
+            }
+        }
+        // Check the output buffer(s) to be sure they are fully initialized.
```

Comment:
Reading upwards, I think these are the output buffers of any extern calls used, rather than output buffers to the pipeline. If that's correct, please clarify in the comment.

Reply:
done